### PR TITLE
ISPN-12918 Rest stats requests block the Netty IO thread

### PR DESCRIPTION
**Pull request by danberindei**
https://issues.redhat.com/browse/ISPN-12918

Process metrics endpoint requests on a blocking executor thread.

### CODE REVIEW

File: server/rest/src/main/java/org/infinispan/rest/resources/MetricsResource.java [hunk 1]
```diff
@@ -67,41 +71,50 @@ public Invocations getInvocations() {
...
-            }
-         } else if (restRequest.method() == OPTIONS) {
-            if (accept.isEmpty()) {
-               accept = Collections.singletonList(APPLICATION_JSON_TYPE);
-            }
-         }
+      List<String> accept = responseMediaTypes(restRequest);
 
+      CompletableFuture<RestResponse> cf = new CompletableFuture<>();
+      blockingExecutor.execute(() -> {
```
Comment (wburns) [inline server/rest/src/main/java/org/infinispan/rest/resources/MetricsResource.java:77]: I would recommend using CompletableFuture.supplyAsync instead so we don't have to do the CompletableFuture complete calls ourselves.

**Review by wburns — COMMENTED** [review]
Just a minor thing. The part I am confused is how this wasn't caught by blockhound... Do we not have a test for this code path ?

**Comment by danberindei** [comment]
Updated @wburns 

I'm not sure why BlockHound didn't catch it, we have an integration test that should touch this API: `RestMetricsResource`

**Comment by wburns** [comment]
Integrated into master, thanks @danberindei !

**Comment by danberindei** [comment]
@wburns found out why BlockHound didn't detect it: the only test for REST metrics is in `server/tests`, where BlockHound isn't enabled. I created ISPN-12918 to add a test in `server/rest`, but I'm not sure who will actually write the test :)

**Pull request merged**